friends/src/components/AddFriend/AddFriend.jsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+
+const AddFriend = props => {
+  const { onAddFriend } = props;
+  const [formInput, setFormInput] = useState({
+    name: "",
+    age: "",
+    email: ""
+  });
+
+  const onSubmitForm = event => {
+    event.preventDefault();
+    const newFriend = {
+      name: formInput.name,
+      age: Number(formInput.age),
+      email: formInput.email
+    };
+
+    onAddFriend(newFriend);
+    setFormInput({
+      name: "",
+      age: "",
+      email: ""
+    });
+  };
+
+  const onChangeInput = event => {
+    event.persist();
```

Comment:
I looked into it. React is doing <a href="https://reactjs.org/docs/events.html#event-pooling">event pooling</a>, that means the event value (a change in the input field in this case), get nullified. This is done for performances reasons.  So when it tries to run line 31 after the first change happened it pass a null to it and the app crash. 
Event.persist() is just made to prevent this to happen.